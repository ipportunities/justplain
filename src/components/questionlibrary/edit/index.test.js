import React from 'react'
import { shallow } from 'enzyme'
import QuestionLibraryEdit from './index'

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn().mockReturnValue({ auth: { token: 'a' } })
}))

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    id: 0,
  }),
}))

describe('Question Library Edit', () => {
  it('should render correctly', () => {
    const component = shallow(<QuestionLibraryEdit />)
    expect(component).toMatchSnapshot()
  })
})
