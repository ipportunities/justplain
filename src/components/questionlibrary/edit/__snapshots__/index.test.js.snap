// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Library Edit should render correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row border-bottom border-info"
  >
    <div
      className="col-6 offset-3"
    >
      <div
        className="row"
      >
        <h1>
          Edit question category
        </h1>
      </div>
      <div
        className="row pb-1 pt-5"
      >
        <form
          className="col-12"
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="categoryName"
            >
              Category name
            </label>
            <input
              className="form-control pl-2 pr-2"
              id="categoryName"
              onChange={[Function]}
              type="text"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="row justify-content-center p-3 border-bottom"
  >
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Add question 
      <i
        className="fa fa-plus"
      />
    </button>
  </div>
</div>
`;
